ckanext/ckanext-apicatalog/ckanext/apicatalog/validators.py

```diff
@@ -368,3 +370,23 @@ def ignore_non_existent_organizations(value):
             pass
 
     return ','.join(existing_organizations)
+
+
+def list_to_json_string(value):
+    if value is not missing:
+        if isinstance(value, str):
+            try:
+                value = json.loads(value)
+            except json.JSONDecodeError:
+                raise toolkit.Invalid('Failed to decode JSON string')
```

Comment:
Shouldn't the value being a string be an error? We're expecting a list here, right? This just checks that the string is JSON and sends it along as-is. I'd expect the output from this validator to be a json-serialized list and nothing else.